### PR TITLE
Add useful string representation for BackendError.

### PR DESCRIPTION
**Pull request by athornton**
Remove unused variables.

Bump version.

### CODE REVIEW

File: apikit/convenience.py [hunk 1]
```diff
@@ -287,6 +282,11 @@ def __init__(self, reason, status_code=None, content=None):
                 raise TypeError("'content' must be a basestring")
         self.content = content
 
+    def __str__(self):
+        """Useful textual representation"""
+        return "BackendError: %d %s [%s]" % (self.status_code,
+                                             self.reason, self.content)
```
Comment (jmatt) [inline apikit/convenience.py:288]: SHAME! But I don't think we have a strong opinion on % versus using format method when manipulating strings.

File: tests/test_backenderror.py [hunk 2]
```diff
@@ -36,3 +36,7 @@ def test_backenderror():
                   "status_code": exc.status_code,
                   "error_content": exc.content}
     assert exc.to_dict() == resultdict
+    # Check string representation
+    expected = "BackendError: %d %s [%s]" % (exc.status_code, exc.reason,
```
Comment (jmatt) [inline tests/test_backenderror.py:40]: format SHAME!

File: apikit/convenience.py [hunk 3]
```diff
@@ -125,12 +125,7 @@ def add_metadata_route(app, route):
         raise TypeError(errstr)
     if not all(isinstance(item, str) for item in route):
         raise TypeError(errstr)
-    name = app.config["NAME"]
     api_version = app.config["API_VERSION"]
-    repository = app.config["REPOSITORY"]
-    version = app.config["VERSION"]
-    description = app.config["DESCRIPTION"]
-    auth = app.config["AUTH"]["type"]
```
Comment (jmatt) [inline apikit/convenience.py:133]: what happened to all these? Were they unused?

**Review by jmatt — APPROVED** [review]

File: apikit/convenience.py (hunk 3 above)
Comment (athornton) [reply apikit/convenience.py:133]: Yeah, I don't know why I had them there in the first place.  flake8 found 'em.

File: tests/test_backenderror.py (hunk 2 above)
Comment (athornton) [reply tests/test_backenderror.py:40]: % rules, format drools.

**Pull request merged**